examples/inference/hybrid_llama.py

```diff
@@ -0,0 +1,96 @@
+import argparse
+import time
+
+import pytest
+import torch
+import torch.distributed as dist
+import transformers
+from transformers import LlamaForCausalLM, LlamaTokenizer
+
+import colossalai
+from colossalai.inference import CaiInferEngine, LlamaModelInferPolicy
+from colossalai.testing import clear_cache_before_run, rerun_if_address_is_in_use, spawn
+
+
+def pipeline_inference_test(args):
+    llama_model_path = args.path
+    max_input_len = args.max_input_len
+    max_output_len = args.max_output_len
+    max_batch_size = args.batch_size
+    micro_batch_size = args.micro_batch_size
+    tp_size = args.tp_size
+    pp_size = args.pp_size
+    rank = dist.get_rank()
+
+    tokenizer = LlamaTokenizer.from_pretrained(llama_model_path)
+    tokenizer.pad_token_id = tokenizer.unk_token_id
+    model = LlamaForCausalLM.from_pretrained(llama_model_path, pad_token_id=tokenizer.eos_token_id)
+    model = model.half()
+
+    model = transformers.LlamaForCausalLM(
+        transformers.LlamaConfig(
+            vocab_size=20000, hidden_size=512, intermediate_size=1536, num_attention_heads=4, num_hidden_layers=4
+        )
+    )
+
+    engine = CaiInferEngine(
```

Comment:
this engine is combining pp and tp?

Reply:
Yes. Now CaiInferEnging support pipeline parallel and tensor parallel.